synapse/app/generic_worker.py

```diff
@@ -975,6 +976,11 @@ def start(config_options):
 
     hs.setup()
 
+    # If background tasks are running on this worker, start collecting the phone
+    # home stats.
+    if hs.config.run_background_tasks:
+        start_phone_stats_home(hs)
```

Comment:
I wonder if we can move this to somewhere in `synapse.app._base`?

Reply:
The only downside of this is that it also gets used for `synapse.app.admin_cmd`. I don't know how frequently this gets run and if it would cause any issues. Thoughts @erikjohnston?

Reply:
Wouldn't the admin command have the `run_background_tasks` disabled?

Reply:
Ah, there's a section to disable all background tasks anyway, so looks like I should add that there.